### DISCUSSION TITLE
Connect internally to KeyCloak?

### DISCUSSION THREAD

**Discussion by tamis-laan** [keycloak/keycloak#22421] [Q&A]
I'm running keycloak inside of a kubernetes cluster using the operator and I'm trying to connect to keycloak locally inside the cluster. Keycloak is exposed on auth.dev.local through the ingress but I'm trying to connect directly using oauth. I'm getting the following error:
```
oidc: issuer did not match the issuer returned by provider, expected "http://keycloak-service:8080/realms/gio/" got "https://auth.dev.local/realms/gio"
```
Is there some way for keycloak to except internal connections? I don't want to go outside the cluster and incur ingress cost for every api<->keycloak call.

**Comment by danielFesenmeyer** [comment]
Maybe it helps to configure different hostnames for frontend and backend? I haven't tried it yet, but it looks like that could help: https://www.keycloak.org/server/hostname

**Reply by tamis-laan** [reply]
Thnx for the tip! I already had the `strictBackchannel` set to `false` in my configuration, thus I would expect it to work. I don't completely understand what it does. Having it set to `true` or `false` doesn't make a difference.

I would expect if there is a backchannel there would be another port opened to connect to.

**Comment by vmuzikar** [comment]
jFTR – relevant issue has been opened: https://github.com/keycloak/keycloak/issues/22667
